### ISSUE TITLE
Debugger: Scrolling on macOS creates blank area

### ISSUE THREAD

**Issue by angstsmurf** [mamedev/mame#11393]
### MAME version

1736d557628f8c8f2a798bb1173d201bab9b9578#diff-cf6805218b4ceea4fc8752d02450b9ec4b7a31d6f8de69097b5a9b1f67e2ac7b

### System information

macOS Ventura 13.4.1, M1 MacBook.

### Emulated system/software

Any

### Incorrect behaviour

Scrolling any view in the debugger creates a blank area instead of showing new information. The new text usually will show up when a key is pressed or the mouse is moved.

### Expected behaviour

Scrolling should reveal new area immediately, as any other program does.

### Steps to reproduce

1. Start MAME in debugger mode.
2. Let it run long enough so that instructions fill the view.
3. Scroll the view.

**Comment by rb6502** [comment]
This is due to how SDL interacts with the native Cocoa event loop in the debugger.  I've been working on and off on a native Cocoa OSD layer which solves this but I haven't yet gotten it to work properly.

**Comment by angstsmurf** [comment]
That is a bummer.

I'm not much of a programmer, but I have some experience with Mac UI coding. If you have any code you are willing to share, I wouldn't mind taking a look at it.

**Comment by angstsmurf** [comment]
Thanks, but I don't think the work-in-progress that @rb6502 is talking about is in there.

**Comment by ksherlock** [comment]
IIRC the issue is that OS/X cocoa scrollviews will ask you to draw an area which is larger than the visible region (and might not even include the currently visible region).  Then as you're scrolling, it doesn't need to ask for a re-draw. But MAME's debug views only draw what it thinks is the currently visible region so other areas can end up blank.

**Comment by rb6502** [comment]
It does scroll beautifully in the OSD=mac build that's pure Cocoa, no SDL, but I've been unable to get the actual graphics output to work there.

@angstsmurf  that's in https://github.com/mamedev/mame/tree/master/src/osd/mac

**Comment by cuavas** [comment]
Fair warning, `OSD=mac` will likely fail to build at the moment.  The src/osd interfaces changed substantially in January, and I kind of updated some of src/osd/mac to match, but with no CI builds, its almost definitely broken right now.

**Comment by rb6502** [comment]
Yeah, I'd meant to look at that before now.  It's really, really broken, to the extent that I'm not sure it's salvageable.  Which is impressive considering there's not that much functionality there to begin with right now.

**Comment by rb6502** [comment]
a8d3327 makes it compile again.  It doesn't start up because for some reason BGFX is forcing OpenGL even though it's being told to use Metal.
